ondevice_learning.qmd

```diff
@@ -1,61 +1,630 @@
+
+This reduction in data transmission not only saves bandwidth but also translates to lower costs for servers, networking, and data storage in the cloud. Large organizations, which might spend millions on cloud infrastructure to train models on-device data, can experience dramatic cost reductions through on-device learning. In the era of Generative AI, where costs have been escalating significantly, finding ways to keep expenses down has become increasingly important. 
+
+Furthermore, the energy and environmental costs associated with running large server farms are also diminished. Data centers are known to consume vast amounts of energy, contributing to greenhouse gas emissions. By reducing the need for extensive cloud-based infrastructure, on-device learning plays a part in mitigating the environmental impact of data processing [@wu2022sustainable].
+
+Specifically for endpoint applications, on-device learning minimizes the number of network API calls needed to run inference through a cloud provider. For applications with millions of users, the cumulative costs associated with bandwidth and API calls can quickly escalate. In contrast, performing training and inferences locally is considerably more efficient and cost-effective. On-device learning has been shown to reduce training memory requirements, drastically improve memory efficiency, and reduce up to 20% in per-iteration latency under the state-of-the-art optimizations [@dhar2021survey].
+
+Another key benefit of on-device learning is the potential for Internet of Things devices to continuously adapt their machine-learning model to new data for continuous, lifelong learning. On-device models can quickly become outdated as user behavior, data patterns, and preferences change. Continuous learning enables the model to efficiently adapt to new data and improvements and maintain high model performance over time.
+
+### Limitations
+
+While traditional cloud-based machine learning systems have access to nearly endless computing resources, on-device learning is often restricted by the limitations in computational and storage power of the edge device that the model is trained on. By definition, an edge device is a device with restrained computing, memory, and energy resources, that cannot be easily increased or decreased. Thus, the reliance on edge devices can restrict the complexity, efficiency, and size of on-device machine learning models. 
+
+#### Compute resources
+
+Traditional cloud-based machine learning systems utilize large servers with multiple high-end GPUs or TPUs that provide nearly endless computational power and memory. For example, services like AWS EC2 allow configuring clusters of GPU instances for massively parallel training. 
```

Comment:
I would write out Amazon Web Services (AWS) here and give a link here to EC2: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/concepts.html